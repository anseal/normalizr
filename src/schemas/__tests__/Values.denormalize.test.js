// eslint-env jest
import { denormalize, schema } from '../../'

describe(`${schema.Values.name} denormalization`, () => {
	test('denormalizes the values of an object with the given schema', () => {
		expect(true).toBe(true)
		const cat = new schema.Entity('cats')
		const dog = new schema.Entity('dogs')
		const valuesSchema = new schema.Values(
			{
				dogs: dog,
				cats: cat,
			},
			(entity, _key) => entity.type
		)

		const entities = {
			cats: { 1: { id: 1, type: 'cats' } },
			dogs: { 1: { id: 1, type: 'dogs' } },
		}

		expect(
			denormalize(
				{
					fido: { id: 1, schema: 'dogs' },
					fluffy: { id: 1, schema: 'cats' },
				},
				valuesSchema,
				entities
			)
		).toMatchSnapshot()
	})
})
