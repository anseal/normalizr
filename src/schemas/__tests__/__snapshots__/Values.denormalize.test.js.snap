// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValuesSchema denormalization denormalizes the values of an object with the given schema 1`] = `
Object {
  "fido": Object {
    "id": 1,
    "type": "dogs",
  },
  "fluffy": Object {
    "id": 1,
    "type": "cats",
  },
}
`;
